src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
```

Comment:
Mmhh, Scala's equals `==` is odd:

```scala
scala> Some(1) == List(1)
<console>:8: warning: comparing values of types Some[Int] and List[Int] using `==` will always yield false
              Some(1) == List(1)
                      ^
res10: Boolean = false

scala> Map(1 -> 'a', 2 -> 'b') == List((1 -> 'a'), (2 -> 'b'))
res11: Boolean = false

scala> (1 -> 'a')
res12: (Int, Char) = (1,a)

scala> Map(1 -> 'a')
res13: scala.collection.immutable.Map[Int,Char] = Map(1 -> a)

scala> Map(1 -> 'a').toList
res14: List[(Int, Char)] = List((1,a))

scala> Map(1 -> 'a').toList == List((1 -> 'a'))
res15: Boolean = true

scala> Map(1 -> 'a') == List((1 -> 'a'))
res16: Boolean = false
```

Javaslang's Option/Some/None is Iterable. Miming Scala's behavior we could compare Iterables in general (e.g. a default equals method in `Iterable` interface). Then Some(1).equals(List(1)) and None.equals(List.empty()) etc.
Or we may provide a method `eq` for that beside `equals`.